frontend/src/modules/processes/reducers.js

```diff
@@ -0,0 +1,140 @@
+import {merge, set} from "object-path-immutable";
+import {map} from "rambda"
+
+import {indexByID} from "../../utils/objects";
+import mergeArray from "../../utils/mergeArray";
+import {summaryReducerFactory} from "../../utils/summary";
+import {templateActionsReducerFactory} from "../../utils/templateActions";
+import {resetTable} from "../../utils/reducers";
+
+import PROCESSES from "./actions";
+
+export const protocols = (
+  state = {
+      items: [],
+      itemsByID: {},
+      isFetching: false,
+  },
+  action
+) => {
+    switch (action.type) {
+        case PROCESSES.LIST_PROTOCOLS.REQUEST:
+            return {
+                ...state,
+                isFetching: true,
+            };
+        case PROCESSES.LIST_PROTOCOLS.RECEIVE:
+            return {
+                ...state,
+                items: action.data,
+                itemsByID: indexByID(action.data, "id"),
+                isFetching: false,
+            };
+        case PROCESSES.LIST_PROTOCOLS.ERROR:
+            return {
+                ...state,
+                isFetching: false,
+                error: action.error,
+            };
+        default:
+            return state;
+    }
+};
```

Comment:
Ideally it should be one file one reducer, I'd put these in their own file with their own action namespace (`PROTOCOLS.LIST` rather than `PROCESSES.LIST_PROTOCOLS`).

Reply:
ok i'll see if i can move them out without blowing something up ;)

Reply:
Ok i moved Protocols into their own files... Sample Kind may need to receive the same treatment if we want this approach to be uniform...

Reply:
Do you intend to use all the code in the newly created protocols files or did you just copy the template from another file? Unless we plan to use all the code, I'd keep only the required minimum.

Reply:
You mean the get and list_table and minimal filter description ? I chose to let them there. This would be more helpful if you could point me to the code you find extraneous ...

Reply:
In `protocols/actions.js`, you only use `.list()`, all the other functions are unused, and the associated parts in the reducer are unused.

Reply:
Thanks i'll clean this up.